source/fundamentals/object-serialization.txt

```diff
@@ -0,0 +1,242 @@
+====================
+Object Serialization
+====================
+
+.. default-domain:: mongodb
+
+Overview
+--------
+
+**Object serialization** is the process of translating an in-memory
+object into a format that can be stored persistently. The reverse process
+is called deserialization. MongoDB stores data as documents containing
+fields with BSON values. The driver uses object
+serialization to translate data between BSON documents and Java objects.
+You can specify how the driver should perform serialization
+using **codecs**.
+
+Codecs implement translation between Extended JSON and Java objects
+according to the ``Codec`` interface using methods that serialize and
+deserialize data. A ``CodecRegistry`` is a class that contains a list
```

Comment:
> A ``CodecRegistry`` is a class that contains a list of codec instances.

This is not quite true, as it contains the providers that contain or dynamically produce the codecs. Perhaps something like:

The ``CodecRegistry`` provides the mechanism that maps Java object types to a specific codec instance.